File: advertising_sdk/src/test/java/com/adadapted/android/sdk/core/zone/ZoneTest.java
package com.adadapted.android.sdk.core.zone;

import com.adadapted.android.sdk.core.ad.Ad;
import com.adadapted.android.sdk.core.common.Dimension;

import org.junit.Test;

import java.util.LinkedList;

import static org.junit.Assert.*;

public class ZoneTest {

    @Test
    public void emptyZone() {
        final Zone zone = Zone.emptyZone();

        assertEquals("", zone.getId());
        assertEquals(0, zone.getDimensions().size());
        assertEquals(0, zone.getAds().size());
        assertFalse(zone.hasAds());
    }

    @Test
    public void builder() {
        final Zone.Builder builder = new Zone.Builder();
        builder.setZoneId("test_zone");
        builder.setDimension(Dimension.Orientation.PORT, new Dimension());
        builder.setDimension(Dimension.Orientation.LAND, new Dimension());
        builder.setAds(new LinkedList<Ad>());

        final Zone zone = builder.build();

        assertEquals("test_zone", zone.getId());
    }
}

File: advertising_sdk/src/test/java/com/adadapted/android/sdk/core/zone/ZoneTest.kt
package com.adadapted.android.sdk.core.zone

import android.util.DisplayMetrics
import com.adadapted.android.sdk.core.view.DimensionConverter
import com.adadapted.android.sdk.core.view.Zone
import org.junit.Assert.assertEquals
import org.junit.Assert.assertFalse
import org.junit.Before
import org.junit.Test

class ZoneTest {
    @Before
    fun setup() {
        val mockDisplayMetrics = DisplayMetrics().apply {
            widthPixels = 1080
            heightPixels = 1920
            density = 3.0f
        }
        DimensionConverter.createInstance(scale = 1f, mockDisplayMetrics)
    }

    @Test
    fun emptyZone() {
        val zone = Zone()
        assertEquals("", zone.id)
        assertEquals(2, zone.dimensions.size.toLong())
        assertEquals(0, zone.ads.size.toLong())
        assertFalse(zone.hasAds())
    }
}